### PR TITLE
Wh/22 create serverapi endpoint for editionsaddons

### PR DESCRIPTION
**Pull request by Rusty12321**
created serverapi endpoint for editionsaddons, and in EditionsAddons.jsx fetched data and reduced redundency.

### CODE REVIEW

File: client/src/components/ProductMain/ProductColumn/EditionsAddons/EditionsAddons.jsx [hunk 1]
```diff
@@ -1,101 +1,72 @@
...
+    fetch("http://localhost:3001/api/editions/1")
+      .then((result) => result.json())
+      .then((data) => setEditions(data));
+
+    fetch("http://localhost:3001/api/add-ons/1")
+      .then((result) => result.json())
+      .then((data) => setAddOns(data));
+  }, []);
+  
+  let display = (item) => {
```
Comment (ranegray) [inline client/src/components/ProductMain/ProductColumn/EditionsAddons/EditionsAddons.jsx:17]: could do a const here, but that's just nitpicking.

File: client/src/components/ProductMain/ProductColumn/EditionsAddons/EditionsAddons.jsx [hunk 2]
```diff
@@ -1,101 +1,72 @@
...
-              </div>
-              <div className="flex items-center px-3">
-                <p className="text-neutral-400">
-                  Unlock exclusive cosmetic items inspired by Han Solo™ and Luke
-                  Skywalker™, as seen in STAR WARS: A New Hope, with Star Wars
-                  Jedi: Survivor Deluxe Edition*!
-                </p>
-              </div>
+            <div className="flex items-center m-2">
+              <a href="https://store.epicgames.com/en-US/p/star-wars-jedi-survivor">
```
Comment (ranegray) [inline client/src/components/ProductMain/ProductColumn/EditionsAddons/EditionsAddons.jsx:36]: urls shouldn't be hardcoded. you might need to refactor the database to include urls.

**Review by ranegray — CHANGES_REQUESTED** [review]
Just a few little things to change and it will be way good.

**Comment by Rusty12321** [comment]
I had to set a max width on the description because once i wrapped the image in an a tag it kept changing the size of the image. Also I wrapped the image in an a tag because the image itself is a link to the page, just like the edition name.

**Review by kittbg — APPROVED** [review]
Looks clean.  No issues.

**Review by ranegray — APPROVED** [review]

**Pull request merged**